NGitLab/IRepositoryClient.cs

```diff
@@ -48,5 +48,9 @@ public interface IRepositoryClient
         IBranchClient Branches { get; }
 
         IProjectHooksClient ProjectHooks { get; }
+
+        CompareInfo Compare(Sha1 from, Sha1 to);
+
+        CompareInfo Compare(string source, string target);
```

Comment:
What if we want to compare a commit against a branch or tag? We should probably always handle `from` and `to` as strings (i.e. provide a single `Compare` method) and let GitLab figure out what they represent.

Furthermore, I think the `Compare` method should take a "query" object instead, to reduce risks of eventual breaking changes if new query parameters are ever introduced:

```cs
public class CompareQuery
{
    /// <summary>
    /// Starting commit SHA or branch name
    /// </summary>
    [JsonPropertyName("from")]
    public string From { get; set; }
	
	/// <summary>
    /// Ending commit SHA or branch name
    /// </summary>
    [JsonPropertyName("to")]
    public string To { get; set; }
}
```